pkg/externaloidc/oidc/externalclaims/tokengetters/clientcredential_test.go

```
@@ -0,0 +1,127 @@
+
+	testcases := []testcase{
+		{
+			name: "nil config, error",
+			cfg:  nil,
+			err:  "client credential configuration must not be nil",
+		},
+		{
+			name: "valid config, no error",
+			tokenHandler: func(w http.ResponseWriter, r *http.Request) {
+				w.Header().Set("Content-Type", "application/json")
+				json.NewEncoder(w).Encode(map[string]interface{}{
```

golangci-lint (2.12.2)

[error] Error return value of `(*encoding/json.Encoder).Encode` is not checked

(errcheck)